blocks/event-speakers/event-speakers.js

```diff
@@ -0,0 +1,255 @@
+
+  // Wait for images to load before checking button states
+  const images = carousel.querySelectorAll('img');
+  if (images.length > 0) {
+    let loadedCount = 0;
+    const checkAllLoaded = () => {
+      loadedCount += 1;
+      if (loadedCount === images.length) {
+        updateNavButtons();
+      }
+    };
+
+    images.forEach((img) => {
+      if (img.complete) {
+        checkAllLoaded();
+      } else {
+        img.addEventListener('load', checkAllLoaded);
+        img.addEventListener('error', checkAllLoaded); // Still check even if image fails
+      }
+    });
+  } else {
+    // No images, check immediately
+    setTimeout(updateNavButtons, 100);
+  }
+}
+
+function reorderSpeakers(speakers, priorityNames) {
+  if (!priorityNames || priorityNames.length === 0) {
+    return speakers;
+  }
+
+  // Normalize priority names for matching (lowercase, trim)
+  const normalizedPriority = priorityNames.map((name) => name.toLowerCase().trim());
+
+  const prioritySpeakers = [];
+  const remainingSpeakers = [];
+
+  // Split speakers into priority and remaining
+  speakers.forEach((speaker) => {
+    const fullName = `${speaker.first_name} ${speaker.last_name}`.toLowerCase();
+    const matchIndex = normalizedPriority.findIndex(
+      (priorityName) => fullName.includes(priorityName) || priorityName.includes(fullName),
+    );
+
+    if (matchIndex !== -1) {
+      prioritySpeakers[matchIndex] = speaker;
+    } else {
+      remainingSpeakers.push(speaker);
+    }
+  });
+
+  // Filter out undefined entries and combine with remaining
+  return [...prioritySpeakers.filter(Boolean), ...remainingSpeakers];
+}
+
+export default async function decorate(block) {
+  // Get the content div (first div in block, standard pattern)
+  const content = block.querySelector('div');
+  if (!content) return;
+
+  // Extract heading and speaker order
+  let headingText = 'Speakers';
+  let speakerOrder = [];
+  const paragraphs = content.querySelectorAll('p');
+
+  let foundHeading = false;
+
+  paragraphs.forEach((p) => {
+    const text = p.textContent.trim();
+    // Skip if it's a link paragraph
+    if (p.querySelector('a')) return;
+
+    // Check if this looks like a speaker order list (contains multiple commas)
+    if (text.includes(',') && text.split(',').length > 2) {
+      speakerOrder = text.split(',').map((name) => name.trim());
+    } else if (!foundHeading && text && !text.includes(',')) {
+      // First non-link text without commas is the heading
+      headingText = text;
+      foundHeading = true;
+    }
+  });
+
+  // Extract API URL
+  let apiUrl = '';
+  const links = content.querySelectorAll('a');
+  links.forEach((link) => {
+    if (link.href.includes('developerevents.adobe.com')) {
+      apiUrl = link.href;
+    }
+  });
+
+  // Extract CTA link (any link that's not the API URL)
+  let ctaHtml = '';
+  links.forEach((link) => {
+    if (!link.href.includes('developerevents.adobe.com')) {
```

Comment:
Unsafe URL validation
```suggestion
    if (new URL(link.href).host !== 'developerevents.adobe.com') {
```